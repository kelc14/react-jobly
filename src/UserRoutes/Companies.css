.Companies-heading {
  margin: 40px auto;
  font-family: "Ubuntu", sans-serif;
  font-size: 3rem;
  font-weight: bold;
  color: goldenrod;
  text-align: center;
  letter-spacing: 5px;
  text-shadow: 2px 7px 5px rgba(0, 0, 0, 0.3),
    0px -4px 10px rgba(255, 255, 255, 0.3);
}
