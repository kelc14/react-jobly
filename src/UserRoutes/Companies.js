import React from "react";
import Company from "./Company";
import "./Companies.css";

const Companies = () => {
  return (
    <div>
      <h2 className="Companies-heading">COMPANIES</h2>
      <Company />
      <Company />
      <Company />
      <Company />
      <Company />
      <Company />
      <Company />
    </div>
  );
};

export default Companies;
