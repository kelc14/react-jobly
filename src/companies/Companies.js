import React, { useEffect, useState } from "react";
import Company from "./Company";
import SearchForm from "../common/SearchForm";
import "./Companies.css";

import JoblyApi from "../api/api";

/** Companies component
 *
 * renders
 * - searchform to filter companies by name
 * - list of all companies (Company components)
 */

const Companies = () => {
  const [companies, setCompanies] = useState([]);

  useEffect(() => {
    const getCompanies = async () => {
      let results = await JoblyApi.getAllCompanies();
      setCompanies(() => [...results]);
    };
    getCompanies();
  }, []);

  const searchCompanies = async (data) => {
    const params = { name: data.search };
    let res = await JoblyApi.getAllCompanies(params);
    setCompanies(() => [...res]);
  };

  return (
    <div>
      <h2 className="Companies-heading">COMPANIES</h2>
      <SearchForm search={searchCompanies} />
      {companies.length === 0 ? (
        <p className="Companies-error">Sorry, no results found.</p>
      ) : (
        companies.map((company) => (
          <Company data={company} key={company.handle} />
        ))
      )}
    </div>
  );
};

export default Companies;
