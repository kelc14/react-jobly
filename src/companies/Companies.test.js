import UserContext from "../hooks/UserContext";
import { render, act } from "@testing-library/react";
import Companies from "./Companies";
import JoblyApi from "../api/api";
import { MemoryRouter } from "react-router-dom";

let testCompanies = {
  companies: [
    {
      handle: "test-company",
      name: "Test Company",
      description: "This is a test company for mocking our API calls.",
      numEmployees: 1000,
      logoUrl: null,
    },
    {
      handle: "test-company2",
      name: "Test Company 2",
      description: "This is another test company for mocking our API calls.",
      numEmployees: 5000,
      logoUrl: null,
    },
  ],
};
/**
 * SMOKE TEST
 */
test("Companies renders without crashing", () => {
  // Mock the getAllCompanies function
  act(() => {
    const getCompaniesMock = jest.fn().mockResolvedValue(testCompanies);

    jest
      .spyOn(JoblyApi, "getAllCompanies")
      .mockImplementation(getCompaniesMock);
  });

  render(
    <MemoryRouter>
      {" "}
      <Companies />
    </MemoryRouter>
  );
});

/**
 * SNAPSHOT TEST
 */
test("Companies matches snapshot", () => {
  // Mock the getAllCompanies function
  act(() => {
    const getCompaniesMock = jest.fn().mockResolvedValue(testCompanies);

    jest
      .spyOn(JoblyApi, "getAllCompanies")
      .mockImplementation(getCompaniesMock);
  });

  const { asFragment } = render(
    <MemoryRouter>
      {" "}
      <Companies />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
