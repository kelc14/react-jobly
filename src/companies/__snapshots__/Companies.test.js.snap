// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Companies matches snapshot 1`] = `
<DocumentFragment>
   
  <div>
    <h2
      class="Companies-heading"
    >
      COMPANIES
    </h2>
    <div
      class="SearchForm"
    >
      <form>
        <input
          class="SearchForm-input"
          id="search"
          name="search"
          placeholder="Enter search term."
          type="text"
          value=""
        />
        <button
          class="SearchForm-btn"
        >
          Search
        </button>
      </form>
    </div>
    <p
      class="Companies-error"
    >
      Sorry, no results found.
    </p>
  </div>
</DocumentFragment>
`;
